utils/self-extracting-executable/decompressor.cpp

```diff
@@ -168,6 +168,10 @@ int decompress(char * input, char * output, off_t start, off_t end, size_t max_n
     return 0;
 }
 
+bool isSudo()
+{
+    return geteuid() == 0 && getenv("SUDO_USER") && getenv("SUDO_UID") && getenv("SUDO_GID"); // NOLINT(concurrency-mt-unsafe)
```

Comment:
What is the reason for checking SUDO_USER? Doesn't `geteuid` cover all the cases?

Reply:
SUDO_ variables make sure that it's sudo and not any other process with setuid - `geteuid` by itself covers too many cases

Reply:
Why should we care if it has been run by root without sudo?

Reply:
well, the first of all if it was run just by root we want resulting file be owned by root - so no chown is needed here, and the second - it's how the task was set - "If clickhouse was run as self-extracting executable under sudo"... but I mostly agree that it doesn't matter actually, and `geteuid` should be enough